lib/common/ipc_client.c

```diff
@@ -510,18 +511,31 @@ pcmk_connect_ipc(pcmk_ipc_api_t *api, enum pcmk_ipc_dispatch dispatch_type)
     }
 
     api->dispatch_type = dispatch_type;
-    switch (dispatch_type) {
-        case pcmk_ipc_dispatch_main:
-            rc = connect_with_main_loop(api);
-            break;
 
-        case pcmk_ipc_dispatch_sync:
-        case pcmk_ipc_dispatch_poll:
-            rc = connect_without_main_loop(api);
+    for (int i = 0; i < n_attempts; i++) {
+        switch (dispatch_type) {
+            case pcmk_ipc_dispatch_main:
+                rc = connect_with_main_loop(api);
+                break;
+
+            case pcmk_ipc_dispatch_sync:
+            case pcmk_ipc_dispatch_poll:
+                rc = connect_without_main_loop(api);
+                break;
+        }
+
+        if (rc == pcmk_rc_ok) {
             break;
-    }
-    if (rc != pcmk_rc_ok) {
-        return rc;
+        }
+        if (rc != EAGAIN) {
+            return rc;
+        }
+
+        /* EAGAIN may occur due to interruption by a signal or due to some
+         * transient issue. Try one more time to be more resilient.
+         */
+        crm_trace("Connection to %s IPC API failed with EAGAIN, retrying",
+                  pcmk_ipc_name(api, true));
```

Comment:
Off-by-one, gonna fix and repush. It would log this message even on the last attempt.